### ISSUE TITLE
OfflinePredictor result different from training result

### ISSUE THREAD

**Issue by yzpick** [tensorpack/tensorpack#1169]
Hi, not sure if this is related to tensorpack or not, stuck here for few days and checked everything... 
I used the same image from the training dataset for the OfflinePredictor and expect the result is at least similar as the result from the training(got from tensorboard). However, the predicted result from tensorboard is good, but the OfflinePredictor gave a bad result, which does not make sense at all. Same input, same model, same parameters, but very different results from training and OfflinePredictor. I am running out of ways to debug. Any ideas for such an issue? I did not have this issue before with the older version of tensorpack, i recently reinstalled the system and newest tensorpack and modified the code accordingly, not sure if i overlooked something...

Thanks!

**Comment by ppwwyyxx** [comment]
Your model can do something different depends on what you write in the models. (e.g., batch norm layers, drop out layers both do something different in training/testing).

**Comment by yzpick** [comment]
I don't think batch norm can cause that much difference I got, I did not expect the exactly same result, but should be similar ones. There is no drop out in the net. uh... it's killing me, the model performs so well in the training and validation, but just goes completely off for the OfflinePredictor. 

Thanks anyway!

**Comment by ganji15** [comment]
@yzpick Have you solved the problem? I meet the same problem too...
